client/blocks/simple-site-rename-form/index.jsx

```diff
@@ -0,0 +1,126 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { get, flow } from 'lodash';
+import Gridicon from 'gridicons';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import FormButton from 'components/forms/form-button';
+import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
+import ConfirmationDialog from './dialog';
+import FormSectionHeading from 'components/forms/form-section-heading';
+import { requestSiteRename } from 'state/site-rename/actions';
+import { isRequestingSiteRename } from 'state/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+export class SimpleSiteRenameForm extends Component {
+	static defaultProps = {
+		currentDomainSuffix: '.wordpress.com',
+		currentDomain: {},
+	};
+
+	state = {
+		showDialog: false,
+		domainFieldValue: '',
+	};
+
+	onConfirm = () => {
+		const { selectedSiteId } = this.props;
+		// @TODO: Give ability to chose whether or not to discard the original domain name.
+		const discard = true;
+
+		this.props.requestSiteRename( selectedSiteId, this.state.domainFieldValue, discard );
+	};
+
+	showConfirmationDialog() {
+		this.setState( {
+			showDialog: true,
+		} );
+	}
+
+	onSubmit = event => {
+		this.showConfirmationDialog();
+		event.preventDefault();
+	};
+
+	onDialogClose = () => {
+		this.setState( {
+			showDialog: false,
+		} );
+	};
+
+	onFieldChange = event => {
+		this.setState( { domainFieldValue: get( event, 'target.value' ) } );
+	};
+
+	render() {
+		const { currentDomain, currentDomainSuffix, isSiteRenameRequesting, translate } = this.props;
+		const currentDomainPrefix = get( currentDomain, 'name', '' ).replace( currentDomainSuffix, '' );
+		const isWPCOM = get( currentDomain, 'type' ) === 'WPCOM';
```

Comment:
Please avoid using "magic strings" - if we ever change this type to a different value, it will be so much harder to update instances like this. Instead, you should use named constants (which we [already have defined](https://github.com/Automattic/wp-calypso/blob/39f364a0ef6f2f7d6f2701a0270d828cced96777/client/lib/domains/constants.js#L7) :)).

Reply:
👌Great call! It did feel icky writing this. I think a bit of laziness had kicked in here 🙊

Reply:
No worries, it's easy to miss stuff like this on a large codebase like ours - there's not a week that goes by that I discover some new nifty, useful helper method, class or component, so I want to spread the knowledge :)

Reply:
Addressed in https://github.com/Automattic/wp-calypso/pull/22661